### PR TITLE
fix(setup-test): remove shell true

### LINKED ISSUE #357: 🛠 Tooling: Find a way around using shell: true in setup-test-e2e.js
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

See discussion in https://github.com/JoshuaKGoldberg/template-typescript-node-package/pull/349/files#r1156109852. We use `shell: true` in a `grep` command in `script/setup-test-e2e.js` because without it, depending on the person running the script, the command either hangs or fails with `no such file or directory`. Fun!

https://github.com/sindresorhus/execa/blob/main/readme.md#shell recommends against using `shell: true`. So it'd be nice if someone could figure out how to get the `setup-test-e2e.js` file working.

### Additional Info

`script/setup-test-e2e.js` is the file run to test that `script/setup.js` works as expected. I generally run the file locally with `pnpm run setup:test`. See https://github.com/JoshuaKGoldberg/template-typescript-node-package#setup for an explanation of why we have a setup command.

### PR DESCRIPTION
**Pull request by navin-moorthy**
## PR Checklist

- [x] Addresses an existing open issue: fixes #357
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Added [`globby`](https://github.com/sindresorhus/globby) package. Reason - https://github.com/sindresorhus/execa/issues/555#issuecomment-1501543612

### CODE REVIEW

**Comment by navin-moorthy** [comment]
Can be merged after https://github.com/JoshuaKGoldberg/template-typescript-node-package/pull/368 which fixes the knip lint check

File: script/setup-test-e2e.js [hunk 1]
```diff
@@ -20,15 +21,16 @@ console.log("New package JSON:", newPackageJson);
 assert.equal(newPackageJson.description, description);
 assert.equal(newPackageJson.name, repository);
 
+const files = await globby(["*.*", "**/*.*"], {
+	gitignore: true,
+	ignoreFiles: ["script/setup.js", "script/setup-test-e2e.js"],
+});
```
[the remark this answers is not in the document]
Comment (navin-moorthy) [reply script/setup-test-e2e.js:27]: But this is needed to use grep with default escaping.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Ah looks like `lint:knip` is failing again. @navin-moorthy I think you'll need to add them back as excludes and file an issue on Knip to support `grep`?

**Comment by navin-moorthy** [comment]
Issue created & changes done.

**Review by JoshuaKGoldberg — APPROVED** [review]
Super, thanks!

**Pull request merged**